---
title: "class14: genome informatics"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## examine asthma SNPs
```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
mxl

```

focus on genotype info as %
```{r}
genotypes <- round(table(mxl [,2]) / nrow(mxl) *100, 2)
genotypes
```
There are `r genotypes [1]`% AA genotypes in this population

##Interpreting Base Qualities in R
```{r}
install.packages("seqinr")
install.packages("gtools")
```
```{r}
library(seqinr)
library(gtools)
phred <- asc(s2c("DDDDCDEDCDDDDBBDDDCC@")) - 33
phred
```

##population scale analysis
```{r}
expr <- read.table("https://bioboot.github.io/bimm143_S18/class-material/
rs8067378_ENSG00000172057.6.txt")

summary(" https://bioboot.github.io/bimm143_S18/class-material/
rs8067378_ENSG00000172057.6.txt")

```

determine expression levels of certain genotype 
```{r}
inds.aa <- expr$geno == "A/A"
summary(expr$exp [inds.aa])
```
```{r}
inds.ag <- expr$geno == "A/G"
summary(expr$exp [inds.ag])

inds.gg <- expr$geno == "G/G"
summary(expr$exp [inds.gg])
```

boxplot values using formula (bc multiple boxes)
```{r}
boxplot(exp ~ geno, data=expr)
```

